src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1815,137 +2352,171 @@ void func_80A153FC(EnGo* this, PlayState* play) {
+
+        case ENGO_ASIDE_STORE:
+            return sMsgScriptGoronAsideStore;
+
+        case ENGO_ASIDE_ELDERSROOM:
+            return sMsgScriptGoronAsideEldersRoom;
+
+        case ENGO_ASIDE_ELDERSSON:
+            return sMsgScriptGoronAsideEldersSon;
+
+        case ENGO_MEDIGORON:
+            return sMsgScriptGoronMedigoron;
+
         default:
-            return D_80A16208;
+            return sMsgScriptGoronAsideStore;
     }
 }
 
-void func_80A157C4(EnGo* this, PlayState* play) {
+/**
+ * Action function for dialog, movement due to dialog, and ending dialog
+ */
+void EnGo_Talk(EnGo* this, PlayState* play) {
     s32 pad;
-    Vec3f sp40;
-    Vec3f sp34;
-
-    if (!func_8010BF58(&this->actor, play, func_80A15684(this, play), this->unk_3D8, &this->unk_28C)) {
-        if ((ENGO_GET_F(&this->actor) != ENGO_F_1) && !(this->unk_390 & 0x200)) {
-            Math_Vec3f_Copy(&sp40, &this->unk_38C->world.pos);
-            Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
-            Math_ApproachS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&sp34, &sp40), 4, 0x2AA8);
+    Vec3f targetPos;
+    Vec3f thisPos;
+
+    if (!func_8010BF58(&this->actor, play, EnGo_GetMsgEventScript(this, play), this->msgEventFunc,
+                       &this->msgScriptResumePos)) {
+        if ((ENGO_GET_TYPE(&this->actor) != ENGO_STRETCHER) && !(this->actionFlags & ENGO_FLAG_ROLLED_UP)) {
+            Math_Vec3f_Copy(&targetPos, &this->attentionTarget->world.pos);
+            Math_Vec3f_Copy(&thisPos, &this->actor.world.pos);
+            Math_ApproachS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&thisPos, &targetPos), 4, 0x2AA8);
         }
-        SubS_FillLimbRotTables(play, this->unk_3CE, this->unk_3C8, ARRAY_COUNT(this->unk_3CE));
+        SubS_FillLimbRotTables(play, this->limbRotTableY, this->limbRotTableZ, ARRAY_COUNT(this->limbRotTableY));
         return;
     }
 
-    if ((ENGO_GET_F(&this->actor) == ENGO_F_5) || (ENGO_GET_F(&this->actor) == ENGO_F_6) ||
-        (ENGO_GET_F(&this->actor) == ENGO_F_7)) {
-        this->unk_3BC = 0;
-        this->unk_390 &= ~0x20;
-        this->unk_3BE = 4;
+    if ((ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_STORE) || (ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_ELDERSROOM) ||
+        (ENGO_GET_TYPE(&this->actor) == ENGO_ASIDE_ELDERSSON)) {
+        this->blinkTimer = 0;
+        this->actionFlags &= ~ENGO_FLAG_EYES_OPEN;
+        this->eyeTexIndex = ENGO_EYETEX_CLOSED2;
     }
 
-    this->unk_390 &= ~0x8;
-    SubS_UpdateFlags(&this->unk_390, 3, 7);
-    this->unk_28C = 0;
-    this->unk_390 |= 0x40;
-    this->actionFunc = this->unk_18C;
+    this->actionFlags &= ~ENGO_FLAG_ENGAGED;
+    SubS_UpdateFlags(&this->actionFlags, 3, 7);
+    this->msgScriptResumePos = 0;
+    this->actionFlags |= ENGO_FLAG_LOST_ATTENTION;
+    this->actionFunc = this->interruptedActionFunc;
 }
 
+/**
+ *  Initialize the Goron, deferring most of the work to the first update cycle.
+ */
```

Comment:
We usually don't add this sort of comment to the main 4 actor functions unless they do something unusual, since they're all used in a similar way for everything.